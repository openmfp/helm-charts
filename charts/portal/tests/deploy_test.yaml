suite: deploy
templates:
  - deploy.yaml
chart:
  version: 1.0.0
  appVersion: 1.0.0
release:
  name: portal
tests:
  - it: matches the snapshot
    values:
      - ../test-values.yaml
    template: deploy.yaml
    asserts:
      - matchSnapshot: {}
