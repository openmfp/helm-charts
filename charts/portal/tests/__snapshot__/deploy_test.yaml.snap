matches the snapshot:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app: portal
      name: portal
    spec:
      revisionHistoryLimit: 3
      selector:
        matchLabels:
          app: portal
      strategy:
        rollingUpdate:
          maxSurge: 5
          maxUnavailable: 0
      template:
        metadata:
          labels:
            app: portal
        spec:
          containers:
            - automountServiceAccountToken: true
              env:
                - name: HTTP_PROTOCOL
                  value: https
                - name: OIDC_CLIENT_ID_PORTAL
                  value: "12345"
                - name: OIDC_CLIENT_SECRET_PORTAL
                  valueFrom:
                    secretKeyRef:
                      key: secret
                      name: portal-client-secret-portal
                - name: DISCOVERY_ENDPOINT_PORTAL
                  value: https://auth.example.com/.well-known/openid-configuration
                - name: BASE_DOMAINS_PORTAL
                  value: portal.example.com,portal2.example.com
                - name: IDP_NAMES
                  value: portal
                - name: FEATURE_TOGGLES
                  value: enableSessionAutoRefresh=true
                - name: PORT
                  value: "8080"
                - name: REGION
                  value: null
                - name: IMAGE_TAG
                  value: 1.0.0
                - name: IMAGE_NAME
                  value: ghcr.io/openmfp/portal
                - name: VALID_WEBCOMPONENT_URLS
                  value: .?
              image: ghcr.io/openmfp/portal:1.0.0
              imagePullPolicy: Always
              name: portal
              ports:
                - containerPort: 8080
                  name: http
              resources:
                limits:
                  cpu: "1"
                  memory: 512Mi
                requests:
                  cpu: 40m
                  memory: 50Mi
              securityContext:
                readOnlyRootFilesystem: true
                runAsNonRoot: true
                seccompProfile:
                  type: RuntimeDefault
              serviceAccountName: portal
          securityContext:
            fsGroup: 2000
            runAsGroup: 3000
            runAsUser: 1000
          serviceAccountName: portal
